pkg/storage/utils/decomposedfs/spaces.go

```diff
@@ -282,7 +282,7 @@ func (fs *Decomposedfs) ListStorageSpaces(ctx context.Context, filter []*provide
 		return nil, errtypes.PermissionDenied(fmt.Sprintf("user %s is not allowed to list spaces of other users", ctxpkg.ContextMustGetUser(ctx).GetId().GetOpaqueId()))
 	}
 
-	checkNodePermissions := !canListAllSpaces && !unrestricted
+	checkNodePermissions := !canListAllSpaces || !unrestricted
```

Comment:
The flag `unrestricted` is ambiguous for me. It is not documented at all in the function. You can only guess what should be unrestricted. Yes, that is possible, but not nice.

Reply:
true
1. we may have to touch this again anyway when adding a user filter, see https://github.com/owncloud/ocis/pull/4072
2. the permissions need to be reworked anyway, as there are better ways than passing an explicit unrestricted flag over the api.

so, I'll not touch this in this PR as we just want to push out the ocis/reva release